[Tan Yu Shing] iP

TYS0n1 opened the pull request:
Individual Project for DUKE program by Tan YU Shing

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

Promethees commented:
Good work!

Promethees commented on src/main/java/Duke.java:
```diff
@@ -3,6 +3,40 @@
+            return data;
+        }
+
+        public void printTask(){
```
Good work!

Promethees commented on src/main/java/Duke.java:
```diff
@@ -35,7 +81,7 @@ public static void main(String[] args) {
         System.out.println(" What can I do for you?");
         System.out.println("____________________________________________________________");
 
```
Should use constants for assigning to these => enhance readability and facilitate coding

hui444 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,219 @@
+        }
+    }
+
+    public static void printMessagae(String message){
```
Should this be printMessage?

hui444 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,219 @@
+        }
+    }
+
+    public static class Deadline extends Task{
```
Maybe separate this into another file? (classes Deadline, Event, and Todo)

hui444 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,219 @@
+            listInputs[taskNumber].setIsDone(true);
+            printDoneStatement(listInputs[taskNumber]);
+        }else{
+            printMessagae(" Invalid task number");
```
I like that you included errors

hui444 reviewed (commented):
I like the intuitive variable names!

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

hui444 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,219 @@
+        printMessage(outputMessage);
+    }
+
+    public static void DoneOperation(String input){
```
Maybe use camelCase for these methods? (lines 126, 144, 151, 166)

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Revert "Signed-off-by: tys <tanyushing@outlook.sg>"

This reverts commit b90e406701131c4bf4696bab0f5551ba20016ae4.

# Conflicts:
#	src/main/java/Duke.java

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
branch created

TYS0n1 committed:
edit 1 for branch

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Merge branch 'branch-Level-5'

TYS0n1 committed:
Minor debugging

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Packaged code branch created

TYS0n1 committed:
Removed static classes

TYS0n1 committed:
debugging and optimization

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Merge branch 'branch-A-Packages'

TYS0n1 committed:
Change list to ArrayList and spelling corrections

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Level 6 delete

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Create file taskList functionality
Read file taskList functionality
Catching errors in taskList functionality

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Added save operation
Added format task function
Added writing list to txt version 1

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Finished adding save operation
Added function: tasks save after successful running
Added comment on how the save operation operates

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Merge branch 'branch-Level-6'

TYS0n1 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/duke/Duke.java

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Added save operation to delete operation

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Debugged save operation

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Jar archived
Changed directory search to hard code address, jar file cannot find directory

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
fixed deadline event conflict and root file issues

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Packaged File I/O

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Added packages for parser(useless) and common (constants messages)

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
finished final packages and optmised duke main class

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
fixed bugs related to encoding and decoding
added date and time for tasks deadline and event
added occur function to find tasks on date specified
other minor bug fixes

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
added find command

Signed-off-by: tys <tanyushing@outlook.sg>

okkhoy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,34 @@
+    private StorageHandler storageHandler;
+    private UserInterface ui;
+
+    public Duke(String filePath){
```
need a space between `)` and `{` here.
Tip: find out how to correct these spacing issues on the IDE automatically!

okkhoy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,34 @@
+
+        while(isRunning == 1){
+            input = in.nextLine();
+            isRunning = parseCommand(input);
```
This doesn't read well. To the reader, it seems like you areo only parsing the command, and then deciding what to do based on the return value.
How to fix: 
- Need to find a more descriptive name for the method.
- Need to find a better place to execute the commands rather than in Parser. So code needs some reorganization.

okkhoy commented on src/main/java/duke/common/Messages.java:
```diff
@@ -0,0 +1,27 @@
+import static duke.storage.StorageHandler.listInputs;
+
+public class Messages {
+    public static final String BYE_MESSAGE = " " + "Bye. Hope to see you again soon!";
+    public static final String SAVED_MESSAGE = " " + "Nice! I have saved your list.";
+    public static final String LIST_HEADER_MESSAGE = " "  + "Here are the tasks in your list:";
+    public static final String DONE_HEADER_MESSAGE = " " + "Nice! I've marked this task as done: ";
+    public static final String DEADLINE_FORMAT_MESSAGE = " " + "Invalid deadline declaration\n" +
+            " " + "deadline {info} /by {date}";
+    public static final String EVENT_FORMAT_MESSAGE = " " + "Invalid event declaration\n" +
+            " " + "event {info} /by {date} {time}";
+    public static final String INVALID_INPUT_MAIN_MESSAGE =
+            " " + "☹ OOPS!!! I'm sorry, but I don't know what that means :-(";
+    public static final String EMPTY_DONE_INPUT_MESSAGE = " " + "☹ OOPS!!! The description of a done cannot be empty.";
+    public static final String EMPTY_TODO_INPUT_MESSAGE = " " + "☹ OOPS!!! The description of a todo cannot be empty.";
+    public static final String EMPTY_DEADLINE_INPUT_MESSAGE =
+            " " + "☹ OOPS!!! The description of a deadline cannot be empty.";
+    public static final String EMPTY_EVENT_INPUT_MESSAGE =
+            " " + "☹ OOPS!!! The description of a event cannot be empty.";
+    public static final String EMPTY_DELETE_INPUT_MESSAGE =
+            " " + "☹ OOPS!!! The description of a delete cannot be empty.";
```
Since each of the constants carry some message text, it is better to name them as `MESSAGE_<something>`
Alternatively, since this class only contains messages, you can eliminate the `MESSAGE` word from the variable name, so that it reads `Messages.BYE`, `Messages.INVALID_INPUT`

okkhoy commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,47 @@
+
+public class Parser {
+    /**
+     * Special condition for each commands done, todo, deadline, event.
```
The introductory line can first tell what this class/method is about.
E.g., `Parses the given input to identify the command`. This can be followed by the description you have.

okkhoy commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,47 @@
+     * marking rubric or algorithm.
+     */
+    public static int parseCommand(String input){
+        if(input.equals("bye")) {
+            printMessage(BYE_MESSAGE);
+            return 0;
+        }else if (input.equals("list")){
+            printList();
+        }else if(input.startsWith("done ") == true) {
+            doneOperation(input);
+        }else if(input.equals("save") == true) {
+            saveOperation();
+            printMessage(SAVED_MESSAGE);
+        }else if(input.startsWith("todo ") == true) {
+            todoOperation(input);
+        }else if(input.startsWith("deadline ") == true) {
+            deadlineOperation(input);
+        }else if(input.startsWith("event ") == true) {
+            eventOperation(input);
+        }else if(input.startsWith("delete ") == true) {
+            deleteOperation(input);
+        }else if(input.equals("done") || input.equals("todo") ||
+                input.equals("deadline") || input.equals("event") ||
+                input.equals("delete")) {
+            printMessage(" ☹ OOPS!!! The description of a " + input + " cannot be empty.");
+        }else {
+            printMessage(INVALID_INPUT_MAIN_MESSAGE);
+        }
```
Spacing around the `if ... else` statements are off. Check how you can get the IDE to do all this for you.

okkhoy commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,47 @@
+        }else {
+            printMessage(INVALID_INPUT_MAIN_MESSAGE);
+        }
+        return 1;
```
Consider replacing the magic literal with a constant with an appropriate name.

okkhoy commented on src/main/java/duke/storage/DukeStorageDecoder.java:
```diff
@@ -0,0 +1,78 @@
+        String line;
+        while (output.hasNext()) {
+            line = output.nextLine();
+            //System.out.println(line);
```
Do not retain commented out code. You can recover the content from Git if necessary later.

okkhoy commented on src/main/java/duke/storage/DukeStorageDecoder.java:
```diff
@@ -0,0 +1,78 @@
+
+    public static void addNewTask(String input){
+        if(input.length() <= 8){
+            System.out.println("Invalid task read.");
+            return;
+        }
+
+        String inputHeader = input.substring(0, 2);
+        if(inputHeader.equals("T ") == true || inputHeader.equals("D ") == true ||
+                inputHeader.equals("E ") == true){
+            //Successful reading of data
+        }else{
+            System.out.println("Invalid task read.");
+            return;
+        }
+
+        String inputDoneData = input.substring(2, 8);
+        if(inputDoneData.equals("| 1 | ") == true || inputDoneData.equals("| 0 | ") == true){
+            //Successful reading of data
+        }else{
+            System.out.println("Invalid task read.");
+            return;
+        }
+
+        String inputData = input.substring(8, input.length());
+
+        if(input.startsWith("T ") == true) {
+            todoOperation("todo " + inputData);
+        }else if(input.startsWith("D ") == true) {
+            deadlineOperation("deadline " + inputData);
+        }else if(input.startsWith("E ") == true) {
+            eventOperation("event " + inputData);
+        }else{
+            System.out.println("Invalid task read.");
+        }
+
+        if(inputDoneData.contains("1")){
+            listInputs.get(listInputs.size() - 1).setIsDone(true);
+        }
+    }
```
There are quite a few magic literals here. I would encourage you to use named constants, at least for the more critical ones.

okkhoy commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,24 @@
+        if(getIsDone() == true){
+            return String.format("[D][✓] %s (by: %s)", getData(), dateDue);
+        }else{
+            return String.format("[D][✗] %s (by: %s)", getData(), dateDue);
```
Consider returning only once; that makes it easy to understand the code.
You can, set the appropriate value using a variable to build the final string.

okkhoy commented on src/main/java/duke/taskList/taskListOperations.java:
```diff
@@ -0,0 +1,151 @@
+import static duke.storage.StorageHandler.createNewFile;
+import static duke.ui.UserInterface.*;
+
+public class taskListOperations {
```
Naming convention for classes uses PascalCase

okkhoy commented on src/main/java/duke/taskList/taskListOperations.java:
```diff
@@ -0,0 +1,151 @@
+import java.io.IOException;
+
+import static duke.storage.StorageHandler.listInputs;
+import static duke.common.Messages.*;
```
Do not use wildcard imports. Import individual classes as necessary.

okkhoy commented on src/main/java/duke/taskList/taskListOperations.java:
```diff
@@ -0,0 +1,151 @@
+        }
+    }
+
+    public static void todoOperation(String input){
```
Try to use verb-form names for methods.  E.g., `createTodo`

okkhoy commented on src/main/java/duke/taskList/taskListOperations.java:
```diff
@@ -0,0 +1,151 @@
+    }
+
+    /**
+     * Save operation triggers for the following actions
```
A better comment would be:
`Saves data on ... toDo, deadline ...`

okkhoy reviewed (commented):
Please pay attention to coding standards and coding conventions.

TYS0n1 committed:
Added java doc comments

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
delete txt

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Merge branch 'branch-Level-8' into branch-Level-9

# Conflicts:
#	src/main/java/duke/common/Messages.java
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/taskList/taskListOperations.java
#	src/main/java/duke/ui/UserInterface.java

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
pushing

TYS0n1 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/storage/DukeStorageEncoder.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/ui/UserInterface.java

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Fixed what ever email wanted me to change.
Added remaining comments.
Minor fixes.

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
push tags

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
ug

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
fix

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
changes after feedback

Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>

TYS0n1 committed:
Signed-off-by: tys <tanyushing@outlook.sg>
